modules/tinymce/src/plugins/wordcount/test/ts/browser/PluginTest.ts

```diff
@@ -85,4 +85,131 @@ describe('browser.tinymce.plugins.wordcount.PluginTest', () => {
+    await pWaitForWordcount(3);
+    editor.setContent('<p>word^word</br>test^</p>');
+    await pWaitForWordcount(3);
+    editor.setContent('<p>word^word</p><p>test^</p>');
+    await pWaitForWordcount(3);
+  });
+
+  it('TINY-8122: Treat ~ as a word break', async () => {
+    const editor = hook.editor();
+    await pWaitForWordcount(0);
+    editor.setContent('<p>~word</p>');
+    await pWaitForWordcount(1);
+    editor.setContent('<p>word~</p>');
+    await pWaitForWordcount(1);
+    editor.setContent('<p>word~word</p>');
+    await pWaitForWordcount(2);
+    editor.setContent('<p>~word~</p>');
+    await pWaitForWordcount(1);
+    editor.setContent('<p>word~2</p>');
+    await pWaitForWordcount(2);
+    editor.setContent('<p>word~word test</p>');
+    await pWaitForWordcount(3);
+    editor.setContent('<p>word~word</br>test~</p>');
+    await pWaitForWordcount(3);
+    editor.setContent('<p>word~word</p><p>test~</p>');
+    await pWaitForWordcount(3);
+  });
+
+  it('TINY-8122: Treat | as a word break', async () => {
+    const editor = hook.editor();
+    await pWaitForWordcount(0);
+    editor.setContent('<p>|word</p>');
+    await pWaitForWordcount(1);
+    editor.setContent('<p>word|</p>');
+    await pWaitForWordcount(1);
+    editor.setContent('<p>word|word</p>');
+    await pWaitForWordcount(2);
+    editor.setContent('<p>|word|</p>');
+    await pWaitForWordcount(1);
+    editor.setContent('<p>word|2</p>');
+    await pWaitForWordcount(2);
+    editor.setContent('<p>word|word test</p>');
+    await pWaitForWordcount(3);
+    editor.setContent('<p>word|word</br>test|</p>');
+    await pWaitForWordcount(3);
+    editor.setContent('<p>word|word</p><p>test|</p>');
+    await pWaitForWordcount(3);
+  });
+
+  it('TINY-8122: Treat № as a word break', async () => {
+    const editor = hook.editor();
+    await pWaitForWordcount(0);
+    editor.setContent('<p>№word</p>');
+    await pWaitForWordcount(1);
+    editor.setContent('<p>word№</p>');
+    await pWaitForWordcount(1);
+    editor.setContent('<p>word№word</p>');
+    await pWaitForWordcount(2);
+    editor.setContent('<p>№word№</p>');
+    await pWaitForWordcount(1);
+    editor.setContent('<p>word№2</p>');
+    await pWaitForWordcount(2);
+    editor.setContent('<p>word№word test</p>');
+    await pWaitForWordcount(3);
+    editor.setContent('<p>word№word</br>test№</p>');
+    await pWaitForWordcount(3);
+    editor.setContent('<p>word№word</p><p>test№</p>');
+    await pWaitForWordcount(3);
+  });
+
+  it('TINY-8122: + should be a word break when not next to number', async () => {
+    const editor = hook.editor();
+    await pWaitForWordcount(0);
+    editor.setContent('<p>+word</p>');
+    await pWaitForWordcount(1);
+    editor.setContent('<p>word+</p>');
+    await pWaitForWordcount(1);
+    editor.setContent('<p>word+word</p>');
+    await pWaitForWordcount(2);
+    editor.setContent('<p>+word+</p>');
+    await pWaitForWordcount(1);
+    editor.setContent('<p>word+2</p>');
```

Comment:
The `+` is next to a number?

Reply:
yeah, this was one of the weird failed cases that cause the word count to be 0